src/Functional/functions.php

```diff
@@ -800,3 +803,37 @@ function sequence($monads)
 {
     return traverse(identity, toTraversable($monads));
 }
+
+/**
+ * filterM :: Monad m => (a -> m Bool) -> [a] -> m [a]
+ *
+ * @param callable $f (a -> m Bool)
+ * @param array|Traversable $collection [a]
+ * @return Monad m [a]
+ */
+function filterM(callable $f, $collection)
```

Comment:
Currying even function that has more that 2 arguments is rather standard in this library.
Thanks to that, code can be written in nicer way.